File: frontend/src/utils/ddlTemplateKeys.ts
/** projectJSON.dataTypeDomains.database[] DDL 模板字段（与后端 DdlTemplateKeys 对齐） */
export const DDL_TEMPLATE_KEYS = [
  'createTableTemplate',
  'updateTableComment',
  'deleteTableTemplate',
  'createIndexTemplate',
  'rebuildTableTemplate',
  'createFieldTemplate',
  'updateFieldTemplate',
  'deleteFieldTemplate',
  'deleteIndexTemplate',
  'createPkTemplate',
  'deletePkTemplate',
] as const;

export type DdlTemplateKey = (typeof DDL_TEMPLATE_KEYS)[number];

export const DDL_TEMPLATE_LABELS: Record<DdlTemplateKey, string> = {
  createTableTemplate: '创建表',
  updateTableComment: '表注释',
  deleteTableTemplate: '删除表',
  createIndexTemplate: '创建索引',
  rebuildTableTemplate: '重建表',
  createFieldTemplate: '添加字段',
  updateFieldTemplate: '修改字段',
  deleteFieldTemplate: '删除字段',
  deleteIndexTemplate: '删除索引',
  createPkTemplate: '创建主键',
  deletePkTemplate: '删除主键',
};

/** 非 SQL 方言（如 JAVA 实体生成）不在 DDL 模板页编辑 */
export function isSqlDialect(code: string | undefined): boolean {
  if (!code) {
    return false;
  }
  return code.toUpperCase() !== 'JAVA';
}

export function editorModeForDialect(code: string): string {
  const norm = code.toLowerCase();
  if (norm.includes('postgres') || norm === 'pg') {
    return 'pgsql';
  }
  if (norm.includes('oracle')) {
    return 'sql';
  }
  if (norm.includes('sqlserver') || norm === 'mssql') {
    return 'sqlserver';
  }
  return 'mysql';
}

File: backend/src/main/java/com/erdonline/erd/util/DdlTemplateKeys.java
package com.erdonline.erd.util;

/**
 * projectJSON.dataTypeDomains.database[] 中方言 DDL 模板的键名。
 */
public final class DdlTemplateKeys {

    public static final String TEMPLATE = "template";
    public static final String CREATE_TABLE = "createTableTemplate";
    public static final String DELETE_TABLE = "deleteTableTemplate";
    public static final String REBUILD_TABLE = "rebuildTableTemplate";
    public static final String CREATE_FIELD = "createFieldTemplate";
    public static final String UPDATE_FIELD = "updateFieldTemplate";
    public static final String DELETE_FIELD = "deleteFieldTemplate";
    public static final String CREATE_PK = "createPkTemplate";
    public static final String DELETE_PK = "deletePkTemplate";
    public static final String CREATE_INDEX = "createIndexTemplate";
    public static final String DELETE_INDEX = "deleteIndexTemplate";
    public static final String UPDATE_TABLE_COMMENT = "updateTableComment";

    /** 模板上下文键 */
    public static final String CTX_ENTITY = "entity";
    public static final String CTX_OLD_ENTITY = "oldEntity";
    public static final String CTX_NEW_ENTITY = "newEntity";
    public static final String CTX_FIELD = "field";
    public static final String CTX_INDEX = "index";
    public static final String CTX_MODULE = "module";
    public static final String CTX_SEPARATOR = "separator";

    /** profile.sqlConfig 缺省值 */
    public static final String DEFAULT_SQL_SEPARATOR = "/*SQL@Run*/";

    private DdlTemplateKeys() {
    }
}
